build-tools/packages/build-cli/src/commands/list.ts

```diff
@@ -24,33 +34,76 @@ export default class ListCommand extends BaseCommand<typeof ListCommand> {
 	static flags = {
 		releaseGroup: releaseGroupFlag({ required: true }),
 		...filterFlags,
+		feed: Flags.string({
+			description:
+				"Filter the resulting packages to those that should be published to a particular npm feed. Use 'official' for public npm.",
+			options: ["official", "internal", "internal-test"],
+			helpGroup: "PACKAGE FILTER",
+			required: false,
+		}),
 		tarball: Flags.boolean({
 			description:
 				"Return packed tarball names (without extension) instead of package names. @-signs will be removed from the name, and slashes are replaced with dashes.",
 			default: false,
 		}),
 	};
 
-	public async run(): Promise<PnpmListEntry[]> {
+	public async run(): Promise<ListItem[]> {
 		const context = await this.getContext();
 		const releaseGroup = context.repo.releaseGroups.get(this.flags.releaseGroup);
+
 		if (releaseGroup === undefined) {
 			this.error(`Can't find release group: ${this.flags.releaseGroup}`, { exit: 1 });
 		}
 
 		const filterOptions = parsePackageFilterFlags(this.flags);
-		const pnpmListResults = await pnpmList(releaseGroup.repoPath);
+		const pnpmListResults: ListItem[] = await pnpmList(releaseGroup.repoPath);
```

Comment:
Nit: should this stay `: PnpmListEntry[]` and next line be `: ListItem[]`? To remove any ambiguity about `pnpmList()` potentially setting `.tarball`.

Reply:
You mean changing the next line to something like:

```ts
const filtered = filterPackages(pnpmListResults as ListItem[], filterOptions)
```

That works but it's a cast vs. a type declaration so I prefer the type declaration. I changed the variable name to reduce the chance of confusion.

Reply:
My point was making it clear that the entries returned by `pnpmList()` will never have `.tarball`, so I was thinking more 

```typescript
const pnpmListResults: PnpmListEntry[] = await pnpmList(releaseGroup.repoPath);
const filtered: ListItem[] = filterPackages(pnpmListResults, filterOptions) ...
```

Which might require updating the `map()` somehow... maybe `.map((item): item is ListItem => {` would do it? I recently learned that you can add type predicates to anonymous functions, but not sure if it must be a function that returns a boolean. If so, maybe adding it on `.filter((item): item is ListItem => {` could work. Either way, nice to have but not a deal breaker for the PR.

Reply:
I changed it to `.filter((item): item is ListItem` but I don't find that clearer. It seems more obvious to me to type the array, not the items in functions called on the array. It seems to "hide" the types, even though it ends up doing the right thing.

Reply:
I'd totally type the array when declaring it, agree on that. I just thought to make that work, we'd need to update the map/filter to explicitly say it was returning that type. Which, now that I think about it, might not even be necessary because the only extra property compared to `PnpmListEntry` is optional.